doc/specs/#16599 - Quick Fix.md

```diff
@@ -0,0 +1,43 @@
+---
+author: Carlos Zamora @carlos-zamora
+created on: 2024-04-01
+last updated: 2024-04-02
+issue id: "#16599"
+---
+# Quick Fix
+## Solution Design
+Quick Fix will be a new UI surface that allows the user to interact with the terminal and leverage the context of a specific command being executed. The UI is discussed in the [UI/UX Design section](#ui/ux-design).
+
+The UI itself will live in the TerminalApp project, whereas any logic populating the UI will live in the TerminalControl project. This allows for any consumer of the TerminalControl project to still have access to these features, but have the ability to represent them in whatever UI they please. To accomplish this, the TerminalControl layer must expose a `IVector<String> GetQuickFixContext(UInt32 Row)` which will retrieve all of the quick fix suggestions at the given `Row`. The returned suggestions will be consumed by the TerminalApp layer, where the `IVector<String>` will be converted to "sendInput" actions and displayed in the quick fix UI.
```

Comment:
Any reason as to why not in the Control project itself? We already have precedence in the right-click context menus for "menus the control owns but the app can extend". That might be easier than necessarily plumbing everything up and down.

Reply:
see:
* https://github.com/microsoft/terminal/blob/main/src/cascadia/TerminalControl/TermControl.xaml#L35-L55
* https://github.com/microsoft/terminal/blob/main/src/cascadia/TerminalControl/TermControl.cpp#L166-L217
* https://github.com/microsoft/terminal/blob/main/src/cascadia/TerminalControl/TermControl.cpp#L3637-L3689
* https://github.com/microsoft/terminal/blob/main/src/cascadia/TerminalApp/TerminalPage.cpp#L1678-L1689
* https://github.com/microsoft/terminal/blob/main/src/cascadia/TerminalApp/TerminalPage.cpp#L4815